### ISSUE TITLE
Frame.from_record_items?

### ISSUE THREAD

**Issue by nekobon** [InvestmentSystems/static-frame#100]
Currently we have `Series.from_items`:

```python
In [10]: def gen_item():
    ...:     for i in range(5):
    ...:         yield str(i), i**2
    ...:

In [11]: sf.Series.from_items(gen_item())

Out[11]:
<Series>
<Index>
0        0
1        1
2        4
3        9
4        16
<<U1>    <int64>
```
and `Frame.from_records`:
```python
In [12]: def gen_record():
    ...:     for i in range(5):
    ...:         yield {'squared': i**2, 'cubed': i**3}
    ...:

In [15]: sf.Frame.from_records(gen_record(), index=(str(i) for i in range(5)))

Out[15]:
<Frame>
<Index>  squared cubed   <<U7>
<Index>
0        0       0
1        1       1
2        4       8
3        9       27
4        16      64
<object> <int64> <int64>
```

In the second example, it would be nice if we can let the generator yield its index as well (i.e. to avoid separate index creation `index=(str(i) for i in range(5)`).

How about to add `Frame.from_record_items`, to add the `Series.from_items`-like behavior on `Frame.from_records`

```python
In [16]: def gen_record_item():
    ...:     for i in range(5):
    ...:         yield str(i), {'squared': i**2, 'cubed': i**3}
    ...:

In [17]: sf.Frame.from_record_items(gen_record_item()) # proposing this method
```
which would return the same Frame as `Out[15]`.

**Comment by flexatone** [comment]
Thanks for this suggestion!  i think the name `from_records_items()` might be more consistent / hierarchical. I will explore the implementation.

**Comment by flexatone** [comment]
This is now implemented as `Frame.from_records_items`, and will be available in the 0.4.1 release. 

```python
In [2]: def gen(): 
   ...:     for i in range(3):  
   ...:         yield f'000{i}', {'squared': i**2, 'cubed': i**3}  
   ...:          
   ...:                                                                                                                                                                                                                 

In [3]: import static_frame as sf                                                                                                                                                                                       

In [4]: sf.Frame.from_records_items(gen())                                                                                                                                                                              
Out[4]: 
<Frame>
<Index> squared cubed   <<U7>
<Index>
0000    0       0
0001    1       1
0002    4       8
<<U4>   <int64> <int64>
```
